repos/system_upgrade/el7toel8/actors/checknis/actor.py

```diff
@@ -0,0 +1,121 @@
+from leapp.actors import Actor
+from leapp.reporting import Report
+from leapp.libraries.common.reporting import report_generic
+from leapp.tags import ChecksPhaseTag, IPUWorkflowTag
+from leapp.libraries.common.reporting import report_with_remediation
+import ipaddress
+import re
+
+def get_line(it):
+    line = next(it)
+
+    if line.endswith("\\\n"):
+        line = line[:-2] + " " + get_line(it)
+
+    return line
+
+class CheckNis(Actor):
+    """
+    Check configuration of ypbind and nsswitch. Address of NIS server cannot be
+    specified by host name if NIS is used to resolve host names.
+    """
+
+    name = "check_nis_nsswitch"
+    consumes = ()
+    produces = (Report,)
+    tags = (IPUWorkflowTag, ChecksPhaseTag)
+
+
+    def nis_in_nsswitch_hosts(self):
+        """
+        Check if hosts are resolved by NIS (nsswitch.conf)
+        """
+
+        try:
+            fp = open("/etc/nsswitch.conf")
+            rawlines = fp.readlines()
+            fp.close()
+        except:
```

Comment:
consider at least catching `OSError` only, or is there any reason to catch everything?
```suggestion
        except OSError:
```

Reply:
There's never a reason for such code to catch more than StandardError